--- bluetooth_orig.py	2016-06-07 14:50:36.972000000 +0200
+++ bluetooth.py	2016-06-07 15:02:12.972000000 +0200
@@ -11,6 +11,7 @@
 import socket,struct,array
 from ctypes import *
 
+from scapy.all import *
 from scapy.config import conf
 from scapy.packet import *
 from scapy.fields import *
@@ -117,9 +118,13 @@
     name = "L2CAP Conn Resp"
     fields_desc = [ LEShortField("dcid",0),
                     LEShortField("scid",0),
-                    LEShortEnumField("result",0,["no_info","authen_pend","author_pend"]),
-                    LEShortEnumField("status",0,["success","pend","bad_psm",
-                                               "cr_sec_block","cr_no_mem"]),
+                    LEShortEnumField("result",0,["success", "pend",
+                                                 "cr_bad_psm", "cr_sec_block",
+                                                 "cr_no_mem", "reserved",
+                                                 "cr_inval_scid", "cr_scid_in_use",
+                                                ]),
+                    LEShortEnumField("status",0,["no_info","authen_pend","author_pend","reserved"
+                                                ]),
                     ]
     def answers(self, other):
         return self.scid == other.scid
